### PR TITLE
fix: handle invalid tokens in refresh token operation

### LINKED ISSUE #3646: Refresh Token operation crashes when passed an invalid token
### Link to reproduction

https://github.com/diesieben07/payload/blob/2fbd4bce5f67bbfbaca5ec7d97e46a923c368e89/test/_community/int.spec.ts#L42-L51

### To Reproduce

Pass an invalid JWT to the `refresh-token` operation. Observe Payload responding with a HTTP 500.

### Describe the Bug

The Refresh Token operation only validates whether you have given it a token. It does not validate whether this token was actually successful in logging in the user. The result of this is that the code assumes that if there is a token, then `request.user` must be set, which is incorrect in the case of an invalid token being passed. This then results in the code trying to access properties on `undefined`, resulting in an exception.

### Payload Version

2.0.4

### Adapters and Plugins

_No response_

### PR DESCRIPTION
**Pull request by diesieben07**
## Description

Fixes the "refresh token" operation to correctly reject invalid tokens instead of crashing with an internal server error.
Fixes #3646 

- [X] I have read and understand the [CONTRIBUTING.md](https://github.com/payloadcms/payload/blob/main/CONTRIBUTING.md) document in this repository.

## Type of change

- [X] Bug fix (non-breaking change which fixes an issue)

## Checklist:

- [X] I have added tests that prove my fix is effective or that my feature works
- [X] Existing test suite passes locally with my changes

### CODE REVIEW

File: packages/payload/src/auth/operations/refresh.ts [hunk 1]
```diff
@@ -57,7 +57,7 @@ async function refresh(incomingArgs: Arguments): Promise<Result> {
     },
   } = args
 
-  if (typeof args.token !== 'string') throw new Forbidden(args.req.t)
+  if (typeof args.token !== 'string' || args.req.user == null) throw new Forbidden(args.req.t)
```
Comment (denolfe) [inline packages/payload/src/auth/operations/refresh.ts:60]: I think something to handle all falsy values would be better. Something like `|| !args.req.user`

**Review by denolfe — CHANGES_REQUESTED** [review]
Just one minor change, then this should be good!

**Comment by diesieben07** [comment]
I don't know why the tests failed. It does not seem related to my changes.

**Review by denolfe — APPROVED** [review]

**Comment by denolfe** [comment]
@diesieben07 I think the CI issue has been fixed on latest. Can you rebase latest main? Looking to merge this after CI passes.

**Comment by diesieben07** [comment]
@denolfe Done, looks like all tests succeeded now.

**Comment by denolfe** [comment]
@diesieben07 Thanks! 👍

**Pull request merged**